validate/validator.py

```diff
@@ -0,0 +1,603 @@
+                elif str(option_value.fixed64) is not "":
+                    return num_template(option_value, field, ".".join([x for x in [name, field.name] if x]), option_value.fixed64)
+                elif str(option_value.sfixed32) is not "":
+                    return num_template(option_value, field, ".".join([x for x in [name, field.name] if x]), option_value.sfixed32)
+                elif str(option_value.sfixed64) is not "":
+                    return num_template(option_value, field, ".".join([x for x in [name, field.name] if x]), option_value.sfixed64)
+                else:
+                    return "raise UnimplementedException()"
+    if field.message_type and not field.containing_oneof:
+        for option_descriptor, option_value in field.GetOptions().ListFields():
+            if option_descriptor.full_name == "validate.rules":
+                if str(option_value.duration) is not "":
+                    return duration_template(option_value, field, ".".join([x for x in [name, field.name] if x]))
+                elif str(option_value.timestamp) is not "":
+                    return timestamp_template(option_value, field, ".".join([x for x in [name, field.name] if x]))
+                elif str(option_value.float) or str(option_value.int32) or str(option_value.int64) or \
+                        str(option_value.double) or str(option_value.uint32) or str(option_value.uint64) or \
+                        str(option_value.bool) or str(option_value.string):
+                    return wrapper_template(option_value, field, ".".join([x for x in [name, field.name] if x]))
+                elif str(option_value.bytes):
+                    return "raise UnimplementedException()"
+                elif str(option_value.message) is not "":
+                    return message_template(option_value, field, ".".join([x for x in [name, field.name] if x]))
+                else:
+                    return "raise UnimplementedException()"
+        if field.message_type.full_name.startswith("google.protobuf"):
+            return ""
+        else:
+            return message_template(None, field, ".".join([x for x in [name, field.name] if x]))
+
+
+def file_template(proto_message):
+    file_tmp = """def validate(p):
+    {%- set accessor = p.DESCRIPTOR -%}
+    {% for option_descriptor, option_value in accessor.GetOptions().ListFields() %}
+        {% if option_descriptor.full_name == "validate.disabled" and option_value %}
+    return None
+        {% endif %}
+    {% endfor %}
+    {% for field in accessor.fields %}
+        {{- rule_type(field) -}}
+    {% endfor %}
+    return None"""
+    return Template(file_tmp).render(rule_type=rule_type, p=proto_message, dir=dir)
+
+
+class UnimplementedException(Exception):
+    pass
+
+class ValidationFailed(Exception):
+    pass
+
+def generate_validate(proto_message):
```

Comment:
let's put this at the very top. 

Could we also make something like a _print_validate(function) which could return the string of the function?